lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -800,3 +831,107 @@ def _post(
                 f"Failed to {self.action} any of {report.count} Elasticsearch documents: {json.dumps(report.report)}"
             )
         return jsonify(summary)
+
+    def _delete(
```

Comment:
I think that `_post` and `_delete` could both be dependent on a new common `ElasticBulkBase` method with all the surrounding boilerplate instead of duplicating it. Also, each subclass has its own `action` for the generate actions helper, and it's true that delete doesn't need the `"doc"`, but I think the common base class can still just call `self.generate_actions` and the specifics can be handled by the generator based on the object context. For example, we'd talked earlier about using the `Schema` for this, with an `attributes` parameter that would be carried through and allow the `DELETE` schema to say it wants the delete action and no doc where the `POST` schema wants update and triggers building the appropriate doc.

This structure would be cleaner and easier to maintain than the current duplication.